frontend/src/metabase/common/components/CollectionPicker/components/CollectionPicker.tsx

```diff
@@ -60,14 +60,12 @@ export const CollectionPickerInner = (
     });
   }, [options.namespace]);
   const path = pathProp ?? defaultPath;
-
+  const id = isValidCollectionId(initialValue?.id) ? initialValue?.id : "root";
   const {
     data: currentCollection,
     error,
     isLoading: loadingCurrentCollection,
-  } = useCollectionQuery({
-    id: isValidCollectionId(initialValue?.id) ? initialValue?.id : "root",
-  });
+  } = useGetCollectionQuery(id != null ? { id } : skipToken);
```

Comment:
There were 2 cases with Collections where hooks from `hooks/entity-framework` would clash with RTK query somehow, and RTK query would end up in forever-loading state. I'm not entirely sure why, but I'm sure replacing the old hooks with RTK query hooks helps.